### PR TITLE
feat(presentational): add List component

### PR DESCRIPTION
**Pull request by agory**

### CODE REVIEW

File: src/web/components/Item/Item.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import Item from './';
+
+describe('Item', () => {
+  let wrapper;
+  let data;
+  let renderItem;
+
+  beforeEach(() => {
+    data = { id: 'GDJETI12K3J4L', name: 'turtle' };
+    renderItem = item => <div>{item.name}</div>;
```
Comment (mfrachet) [inline src/web/components/Item/Item.test.js]: Why to use such a method ?

Comment (agory) [reply src/web/components/Item/Item.test.js]: I don't really know. 😄 
I wanted to test, if the function worked well once in the component.
Do I have to try it with `jest. fn()` and see how many times it is called and the parameters passed through the input ?

File: src/web/components/Item/Item.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('Item', () => {
+  let wrapper;
+  let data;
+  let content;
+
+  beforeEach(() => {
+    data = { id: 'GDJETI12K3J4L', name: 'turtle' };
+    content = <div>{data.name}</div>;
+    wrapper = shallow(<Item content={content} />);
```
Comment (mfrachet) [inline src/web/components/Item/Item.test.js]: Why not to put the content as children instead ?

File: src/web/components/Item/Item.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  });
+
+  it('should render a li with a content created by render func', () => {
```
Comment (mfrachet) [inline src/web/components/Item/Item.test.js]: This test name is a bit weird 😊

File: src/web/components/Item/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Item({ content }) {
```
Comment (mfrachet) [inline src/web/components/Item/index.js]: Why not to use a children here instead ?

File: src/web/components/List/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Item from '../Item';
+
+export default function List({ data, renderItem, keyName }) {
+  if (data.length < 1) {
```
Comment (mfrachet) [inline src/web/components/List/index.js]: you can also do 

```javascript
if (!data.length) {
}
```

File: src/web/components/List/index.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    return null;
+  }
+
+  const content = data.map((item, index) => {
+    const keyValue = keyName ? item[keyName] : index;
```
Comment (mfrachet) [inline src/web/components/List/index.js]: We'll discuss this tonight 😄

File: src/web/components/Item/Item.test.js (hunk 2 above)
Comment (agory) [reply src/web/components/Item/Item.test.js]: I thought it was shorter to write (avoid closing the tag).
However, I can change it

File: src/web/components/Item/Item.test.js (hunk 3 above)
Comment (agory) [reply src/web/components/Item/Item.test.js]: Oh, sorry. It's fix :smile:

File: src/web/components/Item/index.js (hunk 4 above)
Comment (agory) [reply src/web/components/Item/index.js]: I updated the component to use this way

File: src/web/components/List/index.js (hunk 5 above)
Comment (agory) [reply src/web/components/List/index.js]: Thanks I will stop using my noob method :smile:

**Comment by agory** [comment]
Pr is normally ok unless I've done something else wrong.

**Comment by mfrachet** [comment]
What about sublists ?

```html
<ul>
 <li>
  <ul>
     <li></li>
  </ul>
 </li>
</ul>
```

**Comment by agory** [comment]
I thought that to make a sublist the solution would be to create a recursive renderItem method. 
For example for the Menu it would look like this:
```
const renderItem = (item) => {
  if (item.items && item.items.length) {
    return (
      <List key={item.name}>
        <Menu data={item} />
      </List>
    );
  }
  return (
    <Item key={item.name}>
      <Text>{item.name}</Text>
    </Item>
  );
};

export default function Menu({ data }) {
  return (
    <nav>
      <Text className="menu-label">{data.name}</Text>
      <List className="menu-list" data={data.items} renderItem={renderItem} />
    </nav>
  );
}
```
After if it is necessary to integrate the recursive side directly in the List component, I can, but I don't see how I can do that reusably. I will be forced to give a fixed structure to the input of the list and I could no longer give it anything like now.

**Pull request merged**